File: modules/benchmarks_external/src/naive_infinite_scroll/random_data.dart
library random_data;

import 'common.dart';

List<Offering> generateOfferings(int count) =>
    new List.generate(count, generateOffering);

Offering generateOffering(int seed) {
  final res = new Offering();
  res.name = generateName(seed++);
  res.company = generateCompany(seed++);
  res.opportunity = generateOpportunity(seed++);
  res.account = generateAccount(seed++);
  res.basePoints = seed % 10;
  res.kickerPoints = seed % 4;
  res.status = STATUS_LIST[seed % STATUS_LIST.length];
  res.bundles = randomString(seed++);
  res.dueDate = randomDate(seed++);
  res.endDate = randomDate(seed++, minDate: res.dueDate);
  res.aatStatus = AAT_STATUS_LIST[seed % AAT_STATUS_LIST.length];
  return res;
}

Company generateCompany(int seed) {
  return new Company()..name = generateName(seed);
}

Opportunity generateOpportunity(int seed) {
  return new Opportunity()..name = generateName(seed);
}

Account generateAccount(int seed) {
  return new Account()..accountId = seed;
}

String generateName(int seed) {
  const names = const [
    'Foo',
    'Bar',
    'Baz',
    'Qux',
    'Quux',
    'Garply',
    'Waldo',
    'Fred',
    'Plugh',
    'Xyzzy',
    'Thud',
    'Cruft',
    'Stuff'
  ];
  return names[seed % names.length];
}

DateTime randomDate(int seed, {DateTime minDate}) {
  if (minDate == null) {
    minDate = new DateTime.now();
  }

  const offsets = const [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
  return minDate.add(new Duration(days: offsets[seed % offsets.length]));
}

String randomString(int seed) {
  return new String.fromCharCodes(new List.generate(
      const [5, 7, 9, 11, 13][seed % 5],
      (i) =>
          'a'.codeUnitAt(0) + const [0, 1, 2, 3, 4, 5, 6, 7, 8][seed % 9] + i));
}

File: modules/benchmarks/src/naive_infinite_scroll/random_data.ts
import {StringWrapper} from '@angular/facade';
import {
  CustomDate,
  Offering,
  Company,
  Opportunity,
  Account,
  STATUS_LIST,
  AAT_STATUS_LIST
} from './common';

export function generateOfferings(count: number): Offering[] {
  var res = [];
  for (var i = 0; i < count; i++) {
    res.push(generateOffering(i));
  }
  return res;
}

export function generateOffering(seed: number): Offering {
  var res = new Offering();
  res.name = generateName(seed++);
  res.company = generateCompany(seed++);
  res.opportunity = generateOpportunity(seed++);
  res.account = generateAccount(seed++);
  res.basePoints = seed % 10;
  res.kickerPoints = seed % 4;
  res.status = STATUS_LIST[seed % STATUS_LIST.length];
  res.bundles = randomString(seed++);
  res.dueDate = randomDate(seed++);
  res.endDate = randomDate(seed++, res.dueDate);
  res.aatStatus = AAT_STATUS_LIST[seed % AAT_STATUS_LIST.length];
  return res;
}

export function generateCompany(seed: number): Company {
  var res = new Company();
  res.name = generateName(seed);
  return res;
}

export function generateOpportunity(seed: number): Opportunity {
  var res = new Opportunity();
  res.name = generateName(seed);
  return res;
}

export function generateAccount(seed: number): Account {
  var res = new Account();
  res.accountId = seed;
  return res;
}

var names = [
  'Foo',
  'Bar',
  'Baz',
  'Qux',
  'Quux',
  'Garply',
  'Waldo',
  'Fred',
  'Plugh',
  'Xyzzy',
  'Thud',
  'Cruft',
  'Stuff'
];

function generateName(seed: number): string {
  return names[seed % names.length];
}

var offsets = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];

function randomDate(seed: number, minDate: CustomDate = null): CustomDate {
  if (minDate == null) {
    minDate = CustomDate.now();
  }

  return minDate.addDays(offsets[seed % offsets.length]);
}

var stringLengths = [5, 7, 9, 11, 13];
var charCodeOffsets = [0, 1, 2, 3, 4, 5, 6, 7, 8];

function randomString(seed: number): string {
  var len = stringLengths[seed % 5];
  var str = '';
  for (var i = 0; i < len; i++) {
    str += StringWrapper.fromCharCode(97 + charCodeOffsets[seed % 9] + i);
  }
  return str;
}
